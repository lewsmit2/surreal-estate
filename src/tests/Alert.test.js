import React from "react";
import { render, screen } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert component", () => {
  it("does not render an error or a success message if message props is empty", () => {
    const { asFragment } = render(<Alert message="" />);

    const alert = asFragment();
    expect(alert).toMatchSnapshot();
  });

  it("displays an error message", () => {
    render(<Alert message="Error!" />);

    expect(screen.getByText(/Error/).textContent).toBe("Error!");
  });

  it("renders a error message if message and isSuccess are falsy", () => {
    const { asFragment } = render(<Alert message="Error!" success={false} />);

    expect(screen.getByText("Error!")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays a success message", () => {
    render(<Alert message="Success!" success />);

    expect(screen.getByText(/Success/).textContent).toBe("Success!");
  });

  it("renders a success message if message and isSuccess are truthy", () => {
    const { asFragment } = render(<Alert message="Success!" success={true} />);

    expect(screen.getByText("Success!")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
