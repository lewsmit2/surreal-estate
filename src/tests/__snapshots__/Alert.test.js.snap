// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render an error or a success message if message props is empty 1`] = `<DocumentFragment />`;

exports[`renders a error message if message and isSuccess are falsy 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`renders a success message if message and isSuccess are truthy 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!
  </div>
</DocumentFragment>
`;
